pkg/sysfsnet/interfaces_linux.go

```diff
@@ -0,0 +1,60 @@
+package sysfsnet
+
+import (
+	"bufio"
+	"net"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+var rootDir = "/sys/class/net"
+
+type Interface struct {
+	HardwareAddr net.HardwareAddr
+}
+
+func Interfaces() ([]Interface, error) {
```

Comment:
There is nothing wrong with choosing to write your own functions, but you can use the existing package like `github.com/vishvananda/netlink` to find the MAC. Determine by yourself.